### DISCUSSION TITLE
Installation issue in Jetstream2 supercomputer

### DISCUSSION THREAD

**Discussion by Amir1361** [idaholab/moose#20741] [Q&A General] [answered]
Hello,

I installed the MOOSE on the Rocky Linux 8. But when I wanted to run the tests, I got the attached error. 
 [output.txt](https://github.com/idaholab/moose/files/8447448/output.txt)

Could you please help me with this?

Thanks,
Amir

**Comment by GiudGiud** [comment]
Hi Amir

The error messages say `Application not found` so the executable did not build. Could you please report the log of the `make` command?

Guillaume

**Reply by Amir1361** [reply]
Hi Guillaume,

Thank you for your reply. I have attached the log of the 'make' command.

[Moose_test.txt](https://github.com/idaholab/moose/files/8453683/Moose_test.txt)
[my_test.txt](https://github.com/idaholab/moose/files/8453684/my_test.txt)

Thanks,
Amir

**Reply by GiudGiud** [reply]
Ok now you are getting linking errors with Gffortran. Which platform is this on?

**Reply by Amir1361** [reply]
I am using XSEDE Cloud computing. For running jobs parallelly, I need to run the jobs over the Slurm. So I made an instance based on Rocky8, but I cannot install MOOSE correctly.

**Reply by GiudGiud** [reply]
what does this return:
```
echo $FC $CC $CXX
which mpicc; which mpif90
mpicc --version
```

**Reply by Amir1361** [reply]
Here is the output. Thanks,
Amir
[result.txt](https://github.com/idaholab/moose/files/8456256/result.txt)

**Reply by GiudGiud** [reply]
hello

Did you do this?
```

==> For changes to take effect, close and re-open your current shell. <==

Added mamba to /home/rocky/.bashrc

==> For changes to take effect, close and re-open your current shell. <==
```

You need to abide by what mambda tells you. 

Also, jetstream being a supercomputer as you mentioned, the mamba installation is very unoptimized for it. You should install this way instead:
https://mooseframework.inl.gov/getting_started/installation/hpc_install_moose.html

Guillaume

**Reply by Amir1361** [reply]
Here is the output when I followed up on the mamba orders. 
[terminal_text.txt](https://github.com/idaholab/moose/files/8459838/terminal_text.txt)

I already installed Moose on another supercomputer by Anaconda and it works well.

Also, I made another instance and I tried to use the Moose installation for HPC Cluster. Here is the error message while we have python on our system.
[HPC cluster.txt](https://github.com/idaholab/moose/files/8459845/HPC.cluster.txt)

Thanks,
Amir

**Reply by avtarsinghh1991** [reply]
Hey @Amir1361 

I can see the error as

`python: command not found`. I was also having the same issue while installing MOOSE on XSEDE cluster. It has python3 command instead of python. I don't know why.

But, the solution is that you should make .moose_profile bash file and add following text in it

```
PATH=$PATH:$HOME/bin
export PATH
module load gcc python cmake
```
and source the file.

Every time you need to run, you should compile the .moose_profile bash file or add those lines in the main .bash file.

Best
Avtar

**Reply by GiudGiud** [reply]
@Amir1361 
the output of the commands in the conda environment look right, it should install fine now. Just dont do any performance benchmarking off of this and keep in mind you could have performance issues from this install method. 

Please follow Avtar's advice on installing with the HPC-enabled method and let us know how it goes

**Comment by Amir1361** [comment]
@GiudGiud Thank you! I am still get error by running the tests in the Moose by Conda:
[Moose_Conda.txt](https://github.com/idaholab/moose/files/8466164/Moose_Conda.txt)


Thank you @asingh-mit. I put them in the Moose Profile:
[Moose profile.txt](https://github.com/idaholab/moose/files/8466123/Moose.profile.txt)

But again I got the same error:
[HPC cluster.txt](https://github.com/idaholab/moose/files/8466126/HPC.cluster.txt)

Thank you!

**Reply by GiudGiud** [reply]
Would an alias work for setting python ?
`alias python=python3`

**Comment by Amir1361** [comment]
It does not work. Again I have the python error.

**Reply by avtarsinghh1991** [reply]
Put the following in .moose_profile

```
PATH=$PATH:$HOME/bin
export PATH
module load gcc python cmake
export CC=mpicc
export CXX=mpicxx
export F90=mpif90
export F77=mpif77
export FC=mpif90
```

save the file.

Then 
`source .moose_profile`

then try to install

**Reply by avtarsinghh1991** [reply]
After source the file, try to run the command python and see if the cluster is recognizing it.

**Comment by Amir1361** [comment, marked as answer]
Cluster recognizes python now. But I get the same error again. 
[HPC cluster.txt](https://github.com/idaholab/moose/files/8466642/HPC.cluster.txt)

**Reply by GiudGiud** [reply]
hello is this still an issue?

**Reply by Amir1361** [reply]
Hi, 

It was solved. Thanks!
